services/data/src/engine/helpers/validate.ts

```diff
@@ -35,6 +42,11 @@ export const getResourceQueryErrors = (
     if (type === 'delete' && query.data) {
         errors.push("Mutation type 'delete' does not support property 'data'")
     }
+    if (type === 'jsonPatch' && !Array.isArray(query.data)) {
```

Comment:
Could we test against `type === 'update' && !Array.isArray(query.data)` here and assume it's a json-patch operation ?

If that assumption is too weak, are there other introspective checks we can do on the `data` to make sure that we are dealing with a json-patch request body ?

Reply:
I guess we know the shape of a patch array, so we could make sure that the children are sane.

Reply:
yep we could check if `query.data` is an array, whose values are objects with an `op` key containing one of the following values: `"add", "remove", "replace", "move", "copy", "test"` (see https://datatracker.ietf.org/doc/html/rfc6902/#section-4)

Reply:
Nice, that seems like a solid validation to do.